### ISSUE TITLE
[question] Build log file generation in recipe

### ISSUE THREAD

**Issue by KarstenErnert** [conan-io/conan#15108]
### What is your question?

Hello everyone,

I searched for that topic, but I couldn't find any result here.
Is it somehow feasible to generate a log file from the build in the build function of the Conan recipe? My personal attempts here weren't successfull, but I am also not the most skilled Python developer. The idea would be to generate a log file during the `conan create .` so that the log can be archived as part of the package itself.
Does anyone have any ideas here?

BR,
Karsten

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by AbrilRBS** [comment]
Hi @KarstenErnert thanks a lot for your question.

Conan 2 recently released a metadata feature which aims to solve cases just like this one. [This blog post](https://blog.conan.io/2023/10/24/Conan-launches-metadata-files.html) goes into some detail about using in for different usecases, I suggest you read it - feel free to ask any question if needed after you've done so, we're happy to help where needed :)

**Comment by KarstenErnert** [comment]
Hi @RubenRBS  and thanks for the reply,

the article shows nicely how to work with logs, but my problem is rather, that I would like to dump the cmake output into a text file while building. I tried it with redirecting the output stream for the `cmake.build()` but that didn't work.

**Comment by memsharded** [comment]
I think you can use extra arguments to redirect the stdout:

```python
    def build(self):
        cmake = CMake(self)
        cmake.configure(cli_args=[">", "build.log"])
        cmake.build(cli_args=[">>", "build.log"])
```
That will create the "build.log" files from the commands output.

**Comment by memsharded** [comment]
In any case, we might want to improve the UX for logging build logs into files, even if that works, wdyt @RubenRBS ?

**Comment by AbrilRBS** [comment]
Yeah, I think this would synergize pretty nice with the metadata, let's try to push this for some of the next releases if time allows :)

**Comment by AbrilRBS** [comment]
Hi @KarstenErnert we're going to implement an extra argument in the CMake integration to allow for direct stdout redirection to a file, it should be available for the next 2.0.15 Conan release :)
(So it will look something like `cmake.build(redirect_stdout="build.log")`)

**Comment by KarstenErnert** [comment]
Thanks for the support here guys, I really appreciate your help  in this regard.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15150 has been merged, it will be available next 2.0.15 release
